### ISSUE TITLE
Cannot create property 'path' on string 'required'

### ISSUE THREAD

**Issue by bharatpareek** [adaptlearning/adapt_authoring#2245]
> ⚠️ **CORE TEAM EDIT** ⚠️ This was temporarily patched in 0.6.5, but is awaiting the issue to be fixed in mongoose directly. Once fixed there, we can remove the patch.

## Expected Behaviour
Course should be created on clicking save button.

## Actual Behaviour
Getting the following error while saving the course

**"Something went wrong while saving your data.
Please try again."**

**Terminal Output:**

```
TypeError: Cannot create property 'path' on string 'required'
    at /home/ubuntu/adapt_authoring/node_modules/mongoose/lib/schematype.js:948:30
    at Array.forEach (<anonymous>)
    at SchemaString.SchemaType.doValidate (/home/ubuntu/adapt_authoring/node_modules/mongoose/lib/schematype.js:939:19)
    at /home/ubuntu/adapt_authoring/node_modules/mongoose/lib/document.js:1936:9
    at process._tickCallback (internal/process/next_tick.js:61:11)
```

## Versions
  - Authoring Tool Version: 0.6.4
  - Framework Version: v4
  - Node.js Version: v10.15.1
  - MongoDB Version: v4.0.6
  - Operating System: Ubuntu 18.04
  - Browser: Chrome Beta

**Comment by tomgreenfield** [comment]
Update: looks to be an issue introduced in `mongoose@5.14.5`.

Works fine for me if I downgrade to `5.14.4`.

Cheers to @samumist for the hint.

**Comment by tomgreenfield** [comment]
Code change which causes this error: https://github.com/Automattic/mongoose/commit/1e473c6e81e8c0a12ac1a8ae546e40b6ba8843d6#diff-4584a5389a42c94f452f336cc75def60R39.

**Comment by tomgreenfield** [comment]
Update: this has now been fixed in `mongoose@5.5.7`.

**Comment by canstudios-nicolaw** [comment]
@tomgreenfield I think this can be closed?
